### DISCUSSION TITLE
Setting a header to include in all clientside fetch requests

### DISCUSSION THREAD

**Discussion by alex-statsig** [vercel/next.js#48588] [Help] [answered]
### Summary

I'm working on a gradual rollout of a next.js deployment, and want to do best-effort pinning to the previous rollout mid-session. That is, if a user has the website open subsequent "SPA" navigations should stick to the same build if possible. New traffic will be routed to either based on some rollout logic. To do this, I'm considering adding a build_id env variable, and including that in all request headers. I saw via next.config.js I can set a *response* header from the server for all routes, but I'm not sure if there's something similar for clients.

I understand nextjs has built-in logic to do full page reloads if theres a mismatch (https://nextjs.org/docs/deployment#automatic-updates). My concern is during a partial rollout, I don't want every request to be a full page reload (if I partially rollout by some stable/userID this won't be an issue to be fair).

Curious if anyone has tried to do something similar, or has other ideas how to tackle this

**Comment by alex-statsig** [comment, marked as answer]
This was silently fixed in https://github.com/vercel/next.js/pull/63198 by providing a configuration that will be set on every request, thanks @styfle!

Is there a better way to track requests like this in the future? Not ideal to receive no response for a year and have the problem silently solved with no notice

**Reply by styfle** [reply]
Hello,

It wasn't silent, you can see the announcement here: https://vercel.com/blog/version-skew-protection

The Next.js 14.2 blog also mentions skew protection becoming stable here: https://nextjs.org/blog/next-14-2

Its also documented in the Deploying section of the Next.js docs here: https://nextjs.org/docs/app/building-your-application/deploying#version-skew

**Reply by alex-statsig** [reply]
Thanks, yes the last link (in the docs) is where I came across it recently. I don't typically follow the blog posts, and must have missed the footnote in the version notes.

I mainly meant silent with regard to this discussion request though, this post got no attention but there was some nice progress going on from the vercel team. It would be great to see that progress linked here earlier, but perhaps my request was a little too targeted to the solution
